clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -590,10 +700,8 @@ public void handleDisconnection(String destination) {
         }
 
         @Override
-        public void handleCompletedMetadataResponse(RequestHeader requestHeader, long now, AbstractResponse response) {
-            if (!(response instanceof MetadataResponse))
-                throw new IllegalStateException("Unexpected response type in metadata handler: " + response);
-            handleMetadataResponse(requestHeader, (MetadataResponse) response, now);
+        public void handleCompletedMetadataResponse(RequestHeader requestHeader, long now, MetadataResponse response) {
+            handleMetadataResponse(requestHeader, response, now);
```

Comment:
nit: seems we no longer have much use for `handleMetadataResponse`. Maybe we just move its body here?